src/sql/src/plan/lowering.rs

```diff
@@ -1148,6 +1285,39 @@ where
             });
         }
     });
+    // Collect all the outer columns referenced by any CTE referenced by
+    // the inner relation.
+    inner.visit(&mut |e| match e {
+        HirRelationExpr::Get {
+            id: expr::Id::Local(id),
+            ..
+        } => {
+            if let Some(cte_desc) = cte_map.get(id) {
+                let cte_outer_arity = cte_desc.outer_relation.arity();
+                outer_cols.extend(
+                    col_map
+                        .inner
+                        .iter()
+                        .filter(|(_, position)| **position < cte_outer_arity)
+                        .map(|(c, _)| {
+                            // Map the column to a column reference from within the
```

Comment:
I've added a new `fixup` commit trying my best at explaining this. I've only added comments for explaining the +1 and the -1. I don't see how to update the doc comment in `ColumnMap` for `Let` support, since I don't think it changes at all.

Reply:
Indeed nothing has changed, but I am confused as to whether a `Let` is considered to be "early in a record" or "at the end" since depending on the way you look at it, `Let` can be multiple places:
1) From the perspective of the HirRelationExpr rust object, they would be the outermost nodes.
2) In an explain plan, they would be the innermost nodes.
3) They can also be considered to be wherever Get Local is.

Reply:
Or to put it another way, can you add to the doc comment how we determine the level of a Let?

Reply:
The level of a Let is the root level of the sub-query it is defined in. For example, in the following query, all the from `<scope 1>` directly referenced from within the `cte` are at level `1`. 

```
SELECT ... FROM <scope 1> WHERE (WITH cte AS (...) SELECT ... FROM <scope 2> WHERE ...)
```

When processing a CTE level `0` is always empty since the columns projected by the subquery it belongs in are not visible from within the CTE.

Reply:
I think that doc comment belongs in the definition of `HirRelationExpr`. `lowering.rs` is the consumer of that spec.